### DISCUSSION TITLE
how to compress images of different extensions

### DISCUSSION THREAD

**Discussion by omuleanu** [SixLabors/ImageSharp#2646] [Q&A]
I would like to compress images o different formats, ideally for all, but mostly jpeg and png though.
Is there some universal method, like `img.SaveAsync(filePath, 70)`

currently my code is like this:
```
var img = Image.Load(fstream);

if (img.Width > 900)
{
    var neww = 900;
    var newh = neww / img.Width * img.Height;
    img.Mutate(x => x.Resize(neww, newh));
}

await img.SaveAsync(filePath);
```
it looks like for jpeg, I need to do this:
```
await img.SaveAsync(filePath, new JpegEncoder { Quality = 70 });
```
but the `PngEncoder` doesn't have a `Quality` property,

Also I noticed I can save `.png` files using the JpegEncoder, but I'm not sure if this is the most optimal way.

**Comment by ynse01** [comment]
A PNG image is lossless, so it doesn't "compress" the image, it merely writes it to file in a more efficient manor. JPEG however, is a lossy format, which stores only a part of the original information. This is the reason that the `JpegEncoder` has a Quality property to control how much of the original information to keep. The `PngEncoder` doesn't have this capability by definition, as it will always keep all information, hence it doesn't have this property.

Please note that if you change the file extensions, other programs or libraries will not be able to read the image. ImageSharp is one of the few libraries that isn't confused by the file extension to identify the image format.

**Reply by omuleanu** [reply]
I ended up using:
```
await img.SaveAsync(filePath, new JpegEncoder { Quality = 70 });
```
for all image formats, and changing the file name to `.jpg`
the only program that needs to read the image are browsers (Chrome, FF, etc.)
